[15.05] Fix AuthManager.active_authenticators(). Partially re...

nsoranzo opened the pull request:
...vert changes in `auth_conf.xml` syntax.

After PR https://github.com/galaxyproject/galaxy/pull/75 , the `<filter>` tag of
`config/auth_conf.xml` is broken, see https://github.com/galaxyproject/galaxy/pull/75#issuecomment-98194386 .
I don't think that there is a meaningful way to filter on usernames, so I
changed it to filter on email instead of login.
Also:
- fix `AuthManager.check_registration_allowed()` to use both email and username
  since which is used for login depends on the `AuthProvider` used;
- move ldap `<authenticator>` example to `config/auth_conf.xml.sample` .

Pinging @dctrud and @andrewjrobinson for feedback.

nsoranzo committed:
AuthManager.active_authenticators() should filter on email. Partially revert changes in auth_conf.xml syntax.

After https://github.com/galaxyproject/galaxy/pull/75 , the <filter> tag of
config/auth_conf.xml is broken, see https://github.com/galaxyproject/galaxy/pull/75#issuecomment-98194386 .
I don't think that there is a meaningful way to filter on usernames, so I
changed it to filter on email instead of login.
Also:
- fix AuthManager.check_registration_allowed to use both email and username
  since which is used for login depends on the AuthProvider used;
- move ldap <authenticator> example to config/auth_conf.xml.sample .

dtrudg commented:
Looks pretty good to me. Apologies for breaking the filtering without realising that I had!

I can think of a situation where we might want to filter on usernames here, e.g. if username begins with a certain prefix (like ext_) go to a specific plugin pointing at a separate LDAP directory, but that's a real edge case.

nsoranzo commented:
@dctrud If you prefer, I can rework `active_authenticators()` to receive both email and username as separate parameters, it should be easy.

dtrudg commented:
@nsoranzo that would be great if it's just very quick for you, but don't worry too much - I'm speculating when I say we might want to use that in the future. :-)

nsoranzo committed:
Filter also on username in AuthManager.active_authenticators().

Requested by @dctrud. With respect to previous implementation, email and
username are passed as separate parameters.

nsoranzo commented:
@dctrud Added, now it is possible to filter on both email and username.

dtrudg commented:
@nsoranzo Awesome, thanks! :+1:

jmchilton commented:
Awesome - thanks for the fixes @nsoranzo!
